src/fallback.rs

```diff
@@ -840,22 +840,22 @@ impl fmt::Debug for Literal {
 fn token_stream(mut input: Cursor) -> PResult<TokenStream> {
     let mut trees = Vec::new();
     loop {
-        let input_no_ws = skip_whitespace(input);
-        if input_no_ws.rest.len() == 0 {
-            break;
-        }
-        if let Ok((a, tokens)) = doc_comment(input_no_ws) {
+        if let Ok((a, tokens)) = doc_comment(input) {
             input = a;
             trees.extend(tokens);
             continue;
         }
 
-        let (a, tt) = match token_tree(input_no_ws) {
+        let (a, tt) = match token_tree(input) {
             Ok(p) => p,
             Err(_) => break,
         };
         trees.push(tt);
+
         input = a;
+        if input.is_empty() {
+            break;
+        }
```

Comment:
Is this needed? It looks like `input.is_empty()` would only be true at most once in the entire parse so I don't see a way that this would affect performance. If you omit it, the next loop iteration will fail to parse a token_tree and break, which is still correct behavior.